### PR TITLE
HDDS-10470. Populate Maven dependency cache in separate workflow

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Create a separate CI workflow to populate the cache for Maven dependencies from scratch.  This prevents cache inflation that can happen if existing cache is updated with new dependencies: obsolete dependencies are never removed.

The workflow runs on `master` and `ozone-1.4` branches whenever `pom.xml` files are changed.  It also runs on a schedule to avoid cache miss due to expiry.

Existing workflows are updated to only restore the cache, do not create/update it.

https://issues.apache.org/jira/browse/HDDS-10470

## How was this patch tested?

Tested in fork:

- initial commit: https://github.com/adoroszlai/ozone/actions/runs/8170279080/job/22336101638
- no pom change: https://github.com/adoroszlai/ozone/actions/runs/8170585645/job/22337004185
- pom change: https://github.com/adoroszlai/ozone/actions/runs/8170893673/job/22337974520

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -92,10 +92,10 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-pnpm-
       - name: Cache for maven dependencies
-        uses: actions/cache@v4
+        uses: actions/cache/restore@v4
         with:
           path: |
-            ~/.m2/repository
+            ~/.m2/repository/*/*/*
```
Comment (smengcl) [inline .github/workflows/ci.yml:98]: Does this mean caching maven deps 3 layers deep only?

Comment (smengcl) [reply .github/workflows/ci.yml:98]: Got it.

**Review by smengcl — APPROVED** [review]
Thanks @adoroszlai . lgtm

**Comment by adoroszlai** [comment]
Thanks @smengcl for the review.

**Pull request merged**